src/main/java/ai/deepcode/javaclient/core/AnalysisDataBase.java

```diff
@@ -234,6 +235,7 @@ public void updateCachedResultsForFiles(
           allProjectFiles.stream()
               .filter(Objects::nonNull)
               .filter(file -> !mapFile2Suggestions.containsKey(file))
+              .filter(file -> pdUtils.getFileSize(file) > 0 && pdUtils.getFileSize(file) < MAX_FILE_SIZE)
```

Comment:
That check should be not added here but fixed at  `ai.deepcode.javaclient.core.DeepCodeUtilsBase#isSupportedFileFormat` as we need such check in other places too (not while analyzing)

Reply:
Hm, I thought it was the entry point to any analysis, so filtering there seems to make sense. Do we have other entry points I'm not aware of. I considered using `isSupportedFileFormat` and would even consider putting it there, too - if my understanding is correct about entry points into the code analysis.

Reply:
But yes, you are correct - it is better to have it as you say - just looked again!

Reply:
Incoming commit ;)